### PR TITLE
Add swagger for machine learning workspace RP

### PR DESCRIPTION
**Pull request by ragargms**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Review by jhendrixMSFT — CHANGES_REQUESTED** [review]
Can you please address the linter warnings in the [log](https://travis-ci.org/Azure/azure-rest-api-specs/jobs/287250819).

**Comment by salameer** [comment]
Ping @ravbhatnagar on the ARM review :)

File: specification/machinelearning/resource-manager/Microsoft.MachineLearning/2016-04-01/swagger/workspaces.json [hunk 1]
```diff
@@ -0,0 +1,704 @@
...
+            }
+          }
+        ],
+        "x-ms-examples": {
+          "WorkspaceCreate": {
+            "$ref": "../examples/CreateWorkspace.json"
+          }
+        },
+        "responses": {
+          "200": {
```
Comment (ravbhatnagar) [inline specification/machinelearning/resource-manager/Microsoft.MachineLearning/2016-04-01/swagger/workspaces.json]: Is PUT a long running operation?

File: specification/machinelearning/resource-manager/Microsoft.MachineLearning/2016-04-01/swagger/workspaces.json [hunk 2]
```diff
@@ -0,0 +1,704 @@
...
+          "description": "The immutable id associated with this workspace.",
+          "type": "string",
+          "readOnly": true
+        },
+        "creationTime": {
+          "description": "The creation time for this workspace resource.",
+          "type": "string",
+          "readOnly": true
+        },
+        "workspaceAuthorizationToken": {
```
Comment (ravbhatnagar) [inline specification/machinelearning/resource-manager/Microsoft.MachineLearning/2016-04-01/swagger/workspaces.json]: should not be returned in the response of GET/PUT/PATCH. If needed, this should be exposed through a separate POST API

File: specification/machinelearning/resource-manager/Microsoft.MachineLearning/2016-04-01/swagger/workspaces.json [hunk 3]
```diff
@@ -0,0 +1,704 @@
...
+          "description": "The authorization token to access this workspace. This field will only be populated when the response is returned from a PUT/PATCH request.",
+          "type": "string",
+          "readOnly": true
+        },
+        "studioEndpoint": {
+          "description": "The regional endpoint for the machine learning studio service which hosts this workspace.",
+          "type": "string",
+          "readOnly": true
+        },
+        "keyVaultIdentifierId": {
```
Comment (ravbhatnagar) [inline specification/machinelearning/resource-manager/Microsoft.MachineLearning/2016-04-01/swagger/workspaces.json]: Is this the ARM resource ID of the keyvault resource?

**Review by ravbhatnagar — COMMENTED** [review]
Just one major thing about the secrets being returned in the response of PUT. That will need to be fixed.

File: specification/machinelearning/resource-manager/Microsoft.MachineLearning/2016-04-01/swagger/workspaces.json (hunk 3 above)
Comment (ragargms) [reply specification/machinelearning/resource-manager/Microsoft.MachineLearning/2016-04-01/swagger/workspaces.json]: Nope... This is the url of the key (in Azure key vault) which we use to encrypt workspace

File: specification/machinelearning/resource-manager/Microsoft.MachineLearning/2016-04-01/swagger/workspaces.json (hunk 1 above)
Comment (ragargms) [reply specification/machinelearning/resource-manager/Microsoft.MachineLearning/2016-04-01/swagger/workspaces.json]: This is not a long running operation.

File: specification/machinelearning/resource-manager/Microsoft.MachineLearning/2016-04-01/swagger/workspaces.json (hunk 2 above)
Comment (ragargms) [reply specification/machinelearning/resource-manager/Microsoft.MachineLearning/2016-04-01/swagger/workspaces.json]: This does not get populated in the GET request. However, we do return it as part of a PUT and PATCH request. Also, we do have a POST API which return this key. Not sure why returning it in a POST/PATCH response is an issue. Can you specify why having this as part of PUT/PATCH response is not ok? Readers on this resource won't see the key as we do not populate this field in the GET response. Contributors/Owners will be able to see this by using a PUT/POST/PATCH operation.

**Comment by jhendrixMSFT** [comment]
@ravbhatnagar looks like @ragargms has replied to your questions, what are the next steps?

**Comment by ragargms** [comment]
@ravbhatnagar ping...

File: specification/machinelearning/resource-manager/Microsoft.MachineLearning/2016-04-01/swagger/workspaces.json [hunk 4]
```diff
@@ -0,0 +1,704 @@
...
+          "default": {
+            "description": "Error response describing why the operation failed",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    }, 
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MachineLearning/workspaces/{workspaceName}/resyncstoragekeys": {
```
Comment (ravbhatnagar) [inline specification/machinelearning/resource-manager/Microsoft.MachineLearning/2016-04-01/swagger/workspaces.json]: camelcase action name

File: specification/machinelearning/resource-manager/Microsoft.MachineLearning/2016-04-01/swagger/workspaces.json [hunk 5]
```diff
@@ -0,0 +1,704 @@
...
+          "default": {
+            "description": "Error response describing why the operation failed",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MachineLearning/workspaces/{workspaceName}/listworkspacekeys": {
```
Comment (ravbhatnagar) [inline specification/machinelearning/resource-manager/Microsoft.MachineLearning/2016-04-01/swagger/workspaces.json]: camelcase

File: specification/machinelearning/resource-manager/Microsoft.MachineLearning/2016-04-01/swagger/workspaces.json [hunk 6]
```diff
@@ -0,0 +1,704 @@
...
+      "properties": {
+        "userStorageAccountId": {
+          "description": "The fully qualified arm id of the storage account associated with this workspace.",
+          "type": "string"
+        },
+        "ownerEmail": {
+          "description": "The email id of the owner for this workspace.",
+          "type": "string"
+        },
+        "workspaceType": {
```
Comment (ravbhatnagar) [inline specification/machinelearning/resource-manager/Microsoft.MachineLearning/2016-04-01/swagger/workspaces.json]: Is this similar to SKU?

File: specification/machinelearning/resource-manager/Microsoft.MachineLearning/2016-04-01/swagger/workspaces.json (hunk 2 above)
Comment (ravbhatnagar) [reply specification/machinelearning/resource-manager/Microsoft.MachineLearning/2016-04-01/swagger/workspaces.json]: secrets should not be returned in PUT. Soon the Activity Logs in Azure will start logging responses from PUT/GET and we don't want secrets in that stream. If the POST API to get the key is there, then that should be the one that is used. Returning it in PUT does not add any benefit

**Review by ravbhatnagar — COMMENTED** [review]
Looks good besides one blocker - service should not return secrets in the response of GET/PUT/PATCH. Is it just a swagger issue or service is doing it? 
If the service is not returning it, ARM is OK to approve this.

File: specification/machinelearning/resource-manager/Microsoft.MachineLearning/2016-04-01/swagger/workspaces.json (hunk 4 above)
Comment (ragargms) [reply specification/machinelearning/resource-manager/Microsoft.MachineLearning/2016-04-01/swagger/workspaces.json]: fixed

File: specification/machinelearning/resource-manager/Microsoft.MachineLearning/2016-04-01/swagger/workspaces.json (hunk 5 above)
Comment (ragargms) [reply specification/machinelearning/resource-manager/Microsoft.MachineLearning/2016-04-01/swagger/workspaces.json]: fixed

File: specification/machinelearning/resource-manager/Microsoft.MachineLearning/2016-04-01/swagger/workspaces.json (hunk 6 above)
Comment (ragargms) [reply specification/machinelearning/resource-manager/Microsoft.MachineLearning/2016-04-01/swagger/workspaces.json]: Its a little different. Its mostly paid vs free. The rate is fixed for paid workspaces.

File: specification/machinelearning/resource-manager/Microsoft.MachineLearning/2016-04-01/swagger/workspaces.json (hunk 2 above)
Comment (ragargms) [reply specification/machinelearning/resource-manager/Microsoft.MachineLearning/2016-04-01/swagger/workspaces.json]: Fixed

**Comment by jhendrixMSFT** [comment]
@ragargms I didn't see any answer to @ravbhatnagar question about service returning secrets.  If it's not we can merge this change.

**Comment by ragargms** [comment]
@jhendrixMSFT Hi, I created another PR which includes this change and another one (the other change just copies a swagger from private repo to public repo) I addressed Gaurav's comment about returning a secret in the new PR and got a sign-off from Gaurav (ARM). I will close this PR and we can merge that one. Here's the link to the other PR:
https://github.com/Azure/azure-rest-api-specs/pull/1939

**Comment by jhendrixMSFT** [comment]
Closing per your comment.

**Pull request closed without merging**